Network/Network.cpp

```diff
@@ -625,154 +654,277 @@ namespace WPEFramework
             returnResponse(result)
         }
 
-        uint32_t Network::setIPSettings(const JsonObject& parameters, JsonObject& response)
+	uint32_t Network::setIPSettings(const JsonObject& parameters, JsonObject& response)
         {
-            bool result = false;
-            struct in_addr ip_address, gateway_address, mask;
-            struct in_addr broadcast_addr1, broadcast_addr2;
+            JsonObject internal;
+            string interface;
+            string ipversion;
+            bool autoconfig = false;
+            string netmask;
+            string gateway;
+            string ipaddr;
+            string primarydns;
+            string secondarydns;
 
             if ((parameters.HasLabel("interface")) && (parameters.HasLabel("ipversion")) && (parameters.HasLabel("autoconfig")) &&
                 (parameters.HasLabel("ipaddr")) && (parameters.HasLabel("netmask")) && (parameters.HasLabel("gateway")) &&
                 (parameters.HasLabel("primarydns")) && (parameters.HasLabel("secondarydns")))
             {
-                string interface = "";
-                string ipversion = "";
-                bool autoconfig = false;
-                string ipaddr = "";
-                string netmask = "";
-                string gateway = "";
-                string primarydns = "";
-                string secondarydns = "";
+                internal ["interface"] = interface;
```

Comment:
The input parameters are ***NOT*** used. its passing the **_un-inited_** variable for all the variables below.